controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -140,37 +136,35 @@ public String getName() {
 
     @Override
     public CompletableFuture<Void> deleteScope() {
-        return getStreamsInScopeTableName()
-                .thenCompose(tableName -> storeHelper.deleteTable(tableName, true)
-                                                     .thenAccept(v -> log.debug("table deleted {}", tableName)))
+        CompletableFuture<String> streamsInScopeTableNameFuture = getStreamsInScopeTableName();
+        CompletableFuture<String> rgsInScopeTableNameFuture = getReaderGroupsInScopeTableName();
+        CompletableFuture<String> kvtsInScopeTableNameFuture = getKVTablesInScopeTableName();
+        return CompletableFuture.allOf(streamsInScopeTableNameFuture, rgsInScopeTableNameFuture, kvtsInScopeTableNameFuture)
+                .thenCompose(x -> {
+                    String streamsInScopeTableName = streamsInScopeTableNameFuture.join();
+                    String kvtsInScopeTableName = kvtsInScopeTableNameFuture.join();
+                    String rgsInScopeTableName = rgsInScopeTableNameFuture.join();
+                    return CompletableFuture.allOf(storeHelper.deleteTable(streamsInScopeTableName, true),
+                            storeHelper.deleteTable(kvtsInScopeTableName, true), 
+                            storeHelper.deleteTable(rgsInScopeTableName, true))
+                                     .thenAccept(v -> log.debug("tables deleted {} {} {}", streamsInScopeTableName,
+                                             kvtsInScopeTableName, rgsInScopeTableName));
+                })
                 .thenCompose(deleted -> storeHelper.removeEntry(SCOPES_TABLE, scopeName));
     }
 
     @Override
     public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor) {
```

Comment:
We also need a listReaderGroups() API for listing all ReaderGroups inside a scope.

Reply:
we presently have a rest api for listing readergroups in a scope.. it does so by listing all streams in a scope and then filtering those with prefix _RG. 
Perhaps that should be changed to a new api in streammetadatastore which lists them from the rg metadata.. just that it wont list any older rgs.